open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/OpenMetadataAPIGenericHandler.java

```diff
@@ -10940,6 +10940,132 @@ else if (results.size() == 1)
+     * @param searchStringParameterName parameter providing search string
+     * @param resultTypeGUID unique identifier of the type that the results should match with
+     * @param resultTypeName unique name of the type that the results should match with
+     * @param specificMatchPropertyNames name of properties to visit
+     * @param exactValueMatch does the value need to match exactly?
+     * @param forLineage the request is to support lineage retrieval this means entities with the Memento classification can be returned
+     * @param forDuplicateProcessing the request is for duplicate processing and so must not deduplicate
+     * @param serviceSupportedZones list of supported zones for this service
+     * @param sequencingPropertyName should the results be sequenced?
+     * @param startFrom  index of the list to start from (0 for start)
+     * @param pageSize   maximum number of elements to return
+     * @param effectiveTime the time that the retrieved elements must be effective for (null for any time, new Date() for now)
+     * @param methodName calling method
+     *
+     * @return list of beans
+     * @throws InvalidParameterException the userId is null or invalid.
+     * @throws PropertyServerException there is a problem retrieving information from the repositories.
+     * @throws UserNotAuthorizedException the requesting user is not authorized to issue this request.
+     */
+    public List<EntityDetail> getEntitiesByProperty(String       userId,
+                                                    String       searchString,
+                                                    String       searchStringParameterName,
+                                                    String       resultTypeGUID,
+                                                    String       resultTypeName,
+                                                    List<String> specificMatchPropertyNames,
+                                                    boolean      exactValueMatch,
+                                                    boolean      forLineage,
+                                                    boolean      forDuplicateProcessing,
+                                                    List<String> serviceSupportedZones,
+                                                    String       sequencingPropertyName,
+                                                    int          startFrom,
+                                                    int          pageSize,
+                                                    Date         effectiveTime,
+                                                    String       methodName) throws InvalidParameterException,
+                                                                                    PropertyServerException,
+                                                                                    UserNotAuthorizedException {
+        invalidParameterHandler.validateUserId(userId, methodName);
```

Comment:
Can you use the standard generic handler method  getBeansByValue instead of introducing your own? If you have your own you need to keep it in step with any changes in the generic handlers. 

I notice this method returns EntityDetails, would it not be better to work with the Beans in the OMAS. I The beans are the natural concepts for the OMAS to work with and it therefore does not need to be aware of EntityDetail, which is hidden from it by the generic handler.

Reply:
I removed the new method and used and existing one instead.